src/rtree.ml

```diff
@@ -129,6 +129,57 @@ module Make (E : Envelope) (V : Value with type envelope = E.t) = struct
     | a, b -> { max_node_load = t.max_node_load; tree = Node [ a; b ] }
   (* root split *)
 
+  let rec remove_eq' eq = function
+    | Node ns ->
+        let opts, ns' =
+          List.map
+            (fun (e, t) ->
+              let opt, t' = remove_eq' eq t in
+              (opt, (e, t')))
+            ns
+          |> List.split
+        in
+        let opts = List.concat opts in
+        (opts, Node ns')
+    | Leaf es ->
+        let matching, non_matching = List.partition (fun (_, e) -> eq e) es in
+        let elts = List.map snd matching in
+        (elts, Leaf non_matching)
+    | Empty -> ([], Empty)
+
+  let remove_eq t ty e =
```

Comment:
```suggestion
  let remove_eq t e =
```

Note that we don't have to explicitly pass the runtime type into this function. Instead, it has already been made available as an argument to the functor we are currently inside, so we can just pass `V.t` directly to `Repr.equal`.